source/agora/consensus/state/UTXOSet.d

```diff
@@ -82,11 +83,20 @@ abstract class UTXOCache
         {
             auto utxo_hash = UTXO.getHash(tx_hash, idx);
 
-            // Melting UTXOs will melt after 2016 blocks except ones that go to
-            // the CommonsBudget and others will melt at the next block.
-            auto utxo_value = UTXO((melting && output.address != commons_budget) ?
-                height + 2016 : height + 1, tx.type, output);
-            this.add(utxo_hash, utxo_value);
+            // In general, melting UTXOs will be available after 2016 blocks
+            // UTXOs which are towards the Commons Budget are excluded from this
+            // rule, and will be available from the next block.
+            if (melting && output.address != commons_budget)
+                this.add(utxo_hash, UTXO(height + 2016, tx.type, output));
+            // Additionally, if a transaction has two or more outputs,
+            // the first one is allowed to be under the minimum freezing amount.
+            // If it is, it will not be considered frozen.
+            else if (tx.type == TxType.Freeze && tx.outputs.length >= 2 &&
+                     (idx == 0 && output.value < Amount.MinFreezeAmount))
```

Comment:
@Geod24 I'm so happy that it's in a very simple code.
I am sending you a proposal because I have one question, so please review it.
There is a user with 100,000 BOA that is only one UTXO.
He wants to run a node after freezing 40,000 BOA. 
So he froze 40,000 BOA and sent a transaction with about 60,000 BOA to the refund.
The remaining about 60,000 BOA have also been frozen, so if he uses them for payment, others will not be able to use them for a certain period of time.
And if it is used for payment, other people who receive it will not be able to use it for a certain period of time.

Most users may not know that their 100,000 BOA is a single UTXO.
Therefore, I think it would be better to remove the limit on the amount of the refund.
I would like to know first why you added a limit on the amount of the refund for the freeze, and please kindly review my proposal.

Reply:
> I would like to know first why you added a limit on the amount of the refund for the freeze, and please kindly review my proposal.

Because otherwise it's impossible to create a frozen transaction that *does not* have a refund, but has more than one output. If a user wants to freeze only 40k out of > 80k, they will have to first split their UTXOs. The wallet could handle this automatically.
@AndrejMitrovic any thoughts on this ?